Makefile

```diff
@@ -79,8 +79,8 @@ LIBS = -lnosys
 
 LDFLAGS ?= --specs=nano.specs -lc -lgcc $(LIBS) -mcpu=$(CPU_TYPE) -g -gdwarf-2 
 LDFLAGS += -L. -Lcpu/common -L$(CPU_BASE) -T$(CPU_LINK_MEM) -Tlink_sections.ld 
-LDFLAGS += -nostartfiles -Wl,--gc-sections -mthumb -mcpu=$(CPU_TYPE)
-#LDFLAGS += -ffunction-sections -fdata-sections -flto -Os
+LDFLAGS += -nostartfiles -Wl,--gc-sections -mthumb -mcpu=$(CPU_TYPE) -Os
```

Comment:
It does (surprisingly).  Apparently starting with 4.8, additional optimizations can be performed at link time.